src/cpp/core/system/Xdg.cpp

```diff
@@ -135,6 +136,52 @@ FilePath userDataDir(
    );
 }
 
+void verifyUserDirs(
+   const boost::optional<std::string>& user,
+   const boost::optional<FilePath>& homeDir)
+{
+   auto testDir = [](const FilePath& dir, const ErrorLocation& errorLoc)
+   {
+      Error permError;
+      Error error = dir.ensureDirectory();
```

Comment:
I think this might cause trouble later because there is some first-run logic that checks for the existence of these directories to figure out if we need to do certain kinds of initialization (or migration from old folders). Since the pathological case is generally that these directories exist but are unwritable, how would you feel about having this test function do nothing if the directory isn't in place yet?